code/modules/admin/IsBanned.dm

```diff
@@ -46,79 +46,33 @@
-			else
-				log_access("Failed Login: [key] [computer_id] [address] - Banned [.["reason"]]")
-				return .
-
-	else
-		if(!SSdbcore.Connect())
-			var/msg = "Ban database connection failure. Key [ckey] not checked"
-			log_world(msg)
-			message_admins(msg)
-			return
-
-		var/ipquery = ""
-		var/cidquery = ""
-		if(address)
-			ipquery = " OR ip = INET_ATON('[address]') "
-
-		if(computer_id)
-			cidquery = " OR computerid = '[computer_id]' "
-
-		var/datum/DBQuery/query_ban_check = SSdbcore.NewQuery("SELECT IFNULL((SELECT byond_key FROM [format_table_name("player")] WHERE [format_table_name("player")].ckey = [format_table_name("ban")].ckey), ckey), IFNULL((SELECT byond_key FROM [format_table_name("player")] WHERE [format_table_name("player")].ckey = [format_table_name("ban")].a_ckey), a_ckey), reason, expiration_time, duration, bantime, bantype, id, round_id FROM [format_table_name("ban")] WHERE (ckey = '[ckey]' [ipquery] [cidquery]) AND (bantype = 'PERMABAN' OR bantype = 'ADMIN_PERMABAN' OR ((bantype = 'TEMPBAN' OR bantype = 'ADMIN_TEMPBAN') AND expiration_time > Now())) AND isnull(unbanned)")
-		if(!query_ban_check.Execute(async = TRUE))
-			qdel(query_ban_check)
-			return
-		while(query_ban_check.NextRow())
-			var/pkey = query_ban_check.item[1]
-			var/akey = query_ban_check.item[2]
-			var/reason = query_ban_check.item[3]
-			var/expiration = query_ban_check.item[4]
-			var/duration = text2num(query_ban_check.item[5])
-			var/bantime = query_ban_check.item[6]
-			var/bantype = query_ban_check.item[7]
-			var/banid = query_ban_check.item[8]
-			var/ban_round_id = query_ban_check.item[9]
-			if (bantype == "ADMIN_PERMABAN" || bantype == "ADMIN_TEMPBAN")
-				//admin bans MUST match on ckey to prevent cid-spoofing attacks
-				//	as well as dynamic ip abuse
-				if (ckey(pkey) != ckey)
-					continue
-			if (admin)
-				if (bantype == "ADMIN_PERMABAN" || bantype == "ADMIN_TEMPBAN")
-					log_admin("The admin [key] is admin banned (#[banid]), and has been disallowed access")
-					message_admins("<span class='adminnotice'>The admin [key] is admin banned (#[banid]), and has been disallowed access</span>")
-				else
-					log_admin("The admin [key] has been allowed to bypass a matching ban on [pkey] (#[banid])")
-					message_admins("<span class='adminnotice'>The admin [key] has been allowed to bypass a matching ban on [pkey] (#[banid])</span>")
-					addclientmessage(ckey,"<span class='adminnotice'>You have been allowed to bypass a matching ban on [pkey] (#[banid])</span>")
-					continue
-			var/expires = ""
-			if(duration > 0)
-				expires = "\nThe ban is for [DisplayTimeText(duration MINUTES)] and expires on [expiration] (server time)."  //convert from minutes into deciseconds to display the amount of time in days, hours, minutes, ect.
+	if(!SSdbcore.Connect())
+		var/msg = "Ban database connection failure. Key [ckey] not checked"
+		log_world(msg)
+		message_admins(msg)
+		return

```

Comment:
1: This will spam admins on every connection even when the database isn't enabled.

2: DB connection failure will prevent stickyban matching because this is returning. (This is already the case, not your addition) This should be changed to use an else rather then return so the stickyban code can still run. (Note, stickyban matching is intentionally done last as the stickyban reasons tend to have less info than db bans)